openhcl/underhill_core/src/worker.rs

```diff
@@ -1246,12 +1246,89 @@ async fn new_underhill_vm(
 
     let boot_info = runtime_params.parsed_openhcl_boot();
 
+    // Determine if x2apic is supported so that the topology matches
```

Comment:
Are these changes indeed part of your commit? Or stale from a fork?

Reply:
I have to move these codes up so we can calculate device_dma using processor_topology.vp_count() below.